src/components/Dialog/context.ts

```diff
@@ -21,7 +21,8 @@ export function useDialogControl(): DialogOuterProps['control'] {
     open: () => {},
     close: () => {},
   })
-  const {activeDialogs} = useDialogStateContext()
+  const {activeDialogs, openDialogs} = useDialogStateContext()
+  const isOpen = openDialogs.includes(id)
```

Comment:
Since closing is ultimately handled by the `BottomSheet`, and we need to have a top-level reference to open dialogs for a11y prop setting at the `shell` level, I'm reusing the global state here for this localized reference.